{
  _done.await();
  if (_completed)   return _context;
  if (_cause instanceof CancellationException)   throw (CancellationException)new CancellationException().initCause(_cause);
  throw new ExecutionException(_cause);
}
